### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by Yerimah**
In this branch, the following task were completed:

☑️ Login page and registration page:
   - built with Devise.
   - 
☑️ Food list:

   - displays a list of food.
   - leads to the form that allows users to add new food.
   - 
☑️ Recipies list:

   - displays a list of recipes created by the logged-in user.
   - leads to recipe details.
If the user is the owner of the recipe, should allow user to delete it.

☑️ Public recipe list:

   - displays a list of all public recipes ordered by newest.
   - leads to recipe details.
   - 
☑️ Recipe details:

   - If the recipe is public or the user is the owner of the recipe, it displays the recipe details.
   - If the user is the owner of the recipe, it leads to the form that allows user to add new food.

☑️ General shopping list view:
   - show the list of food that is missing for all recipes of the logged-in user.
   - count the total food items and total price of the missing food.

☑️ Create navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Yerimah and @BenjaminSemah 

Good job so far! I know you have worked hard on making this milestone. There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Required Changes :recycle:
- [ ] Check the comments under the review too.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!:clap::clap::clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/foods/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            <%= f.label :price, "Unit price($) *" %>
+            </div
+
+            <div class="form-group mt-3">
+            <%= f.submit "Add Food", class: "btn", style: "background-color: var(--primary-color); color: var(--white);", data: {disable_with: "Loading..."} %>
+            </div>
+          <% end %>
+        </div>
+      </div>
+    </div>
```
[the remark this answers is not in the document]
Comment (BenjaminSemah) [reply app/views/foods/new.html.erb]: Resolved ✅

File: app/views/recipe_foods/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    <p>Select Ingredient</p>
+    <%= f.select :food_id, @foods.map { |food| [food.name, food.id] }, {}, { class:'form-select'} %>
+
+    <br>
+    <p>Quantity</p>
+    <%= f.number_field :quantity, { class:'form-select'} %>
+
+    <%= f.submit "Add Ingredient" %>
+  <% end %>
+</div>
```
[the remark this answers is not in the document]
Comment (BenjaminSemah) [reply app/views/recipe_foods/new.html.erb:16]: Resolved ✅

File: spec/models/food_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      @food.price = 'Ten'
+      expect(@food).to_not be_valid
+    end
+
+    it 'check if price is greater or equal to zero' do
+      @food.price = -2
+      expect(@food).to_not be_valid
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (BenjaminSemah) [reply spec/models/food_spec.rb:29]: Resolved ✅ 
We have added additional tests to cover more features.

**Comment by BenjaminSemah** [comment]
Hello.

We have worked on the following as pointed out by the previous reviewer.

✅ Fix issues with `Add Food` feature.
✅ Fix issues with `Add Ingredient` feature.
✅ Added additional tests to cover more features.

**Review by Deepakdanger — APPROVED** [review]
**Hi @Team**, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by BenjaminSemah** [comment]
Thanks! 👍

**Pull request merged**